README.md

```diff
@@ -26,6 +26,79 @@ Visit http://localhost:9116/metrics?address=1.2.3.4 where 1.2.3.4 is the IP of t
 SNMP device to get metrics from. You can also specify a `module` parameter, to
 choose which module to use from the config file.
 
+
+## Auththentication
+
+The default configuration is to use SNMPv2 with the community `public`.  This can be changed in the yaml config.  NOTE: Version 2 implies SNMP version 2c.
+
+###Authentication parameters
+
+Name | Description
+--------|------------
+community | Community string defined on the device
+
+Example:
+```YAML
+default:
+  version: 2
+  auth:
+    community: SomeCommunityString
+  walk:
+    - ...
+  metrics:
+    - ...
+```
+
+For SNMPv1, the authentication also requires a community string which will default to 'public'.
+
+###Authentication parameters
+
+Name | Description
+--------|-----------
+community | Community string defined on the device
+
+Example:
+````YAML
+default:
+  version: 1
+  auth:
+    community: SomeCommunityString
+  walk:
+    - ...
+  metrics:
+    - ...
+````
+
+For SNMPv3, the authentication requires different parameters.  The `auth_protocol` defaults to `MD5` and the `priv_protocol` defaults to `DES`.  The `security_level` defaults to `noAuthNoPriv`.
+
+###Authentication parameters
+
+Name | Description | required
+--------|--------------|--------------
+username | A string representing the name of the user | yes
+password |  If messages sent on behalf of this user can be authenticated, the (private) authentication key for use with the authentication protocol. Defined as authKey in RFC3414 | yes
+auth_protocol | An indication of whether messages sent on behalf of this user can be authenticated, and if so, the type of authentication protocol which is used. 2 protocols are defined in RFC3414: MD5 (HMAC-MD5-96) and SHA (HMAC-SHA-96) | no
+priv_protocol | An indication of whether messages sent on behalf of this user can be protected from disclosure, and if so, the type of privacy protocol which is used. Only one protocol is defined in RFC3414: DES (CBC-DES Symmetric Encryption Protocol) | no
+security_level | The Level of Security from which the User-based Security module determines if the message needs to be protected from disclosure and if the message needs to be authenticated. | no (if this is changed, required depends on this)
```

Comment:
security_level determines if priv_password is required or not. At it's default value NoAuthNoPriv no authentication or privacy protocol is used so priv_password is not required